Src/Couchbase/IClusterHelper.cs

```diff
@@ -0,0 +1,116 @@
+using Couchbase.Authentication;
+using Couchbase.Configuration.Client;
+using Couchbase.Core;
+
+#if NET45
+using Couchbase.Configuration.Client.Providers;
+#endif
+
+namespace Couchbase
+{
+    /// <summary>
+    /// A helper object for working with a <see cref="Cluster"/> instance.
+    /// </summary>
+    /// <remarks>Call <see cref="Initialize()"/> before calling <see cref="Get()"/> to get the instance.</remarks>
+    public interface IClusterHelper
+    {
+        /// <summary>
+        /// True if the <see cref="IClusterHelper"/> has been initialized. Calling
+        /// <see cref="Close()"/> will reset this value to false.
+        /// </summary>
+        bool Initialized { get; }
+
+        /// <summary>
+        /// Returns an instance of the Cluster class.
+        /// </summary>
+        /// <remarks>
+        /// Call one of the Initialize() overloads to create or recreate the instance.
+        /// However, Initialize() should only be called when the process starts up.
+        /// </remarks>
+        /// <returns>An instance of the Cluster class.</returns>
+        Cluster Get();
+
+        /// <summary>
+        /// Opens or gets an <see cref="IBucket"/> instance from the <see cref="ICluster"/> that this <see cref="IClusterHelper"/> is wrapping.
+        /// The <see cref="IBucket"/> will be cached and subsequent requests for a <see cref="IBucket"/> of the same name will return the
+        /// cached instance.
+        /// </summary>
+        /// <param name="bucketName">The name of the <see cref="IBucket"/> to open or get.</param>
+        /// <returns>An <see cref="IBucket"/>instance</returns>
+        /// <remarks>Before calling you must call <see cref="IClusterHelper.Initialize()"/>.</remarks>
+        IBucket GetBucket(string bucketName);
+
+        /// <summary>
+        /// Opens or gets an <see cref="IBucket"/> instance from the <see cref="ICluster"/> that this <see cref="IClusterHelper"/> is wrapping.
+        /// The <see cref="IBucket"/> will be cached and subsequent requests for a <see cref="IBucket"/> of the same name will return the
+        /// cached instance.
+        /// </summary>
+        /// <param name="bucketName">The name of the <see cref="IBucket"/> to open or get.</param>
+        /// <param name="password">Bucket password, or null for unsecured buckets.</param>
+        /// <returns>An <see cref="IBucket"/>instance</returns>
+        /// <remarks>Before calling you must call <see cref="IClusterHelper.Initialize()"/>.</remarks>
+        IBucket GetBucket(string bucketName, string password);
+
+        void RemoveBucket(string bucketName);
```

Comment:
@dotarj Seems like you missed a summary here. 

@MikeGoldsmith Would you be interested in having [StyleCop](https://github.com/DotNetAnalyzers/StyleCopAnalyzers) added to the project to make sure this does not happen?